pyam/core.py

```diff
@@ -261,34 +261,43 @@ def metadata(self, meta, name=None):
             (by `['model', 'scenario']` index if possible)
         name: str
             category column name (if not given by meta pd.Series.name)
+        index: pd.MultiIndex
+            index to be used for setting metadata column
         """
-        if isinstance(meta, pd.Series) and \
-                set(['model', 'scenario']).issubset(meta.index.names):
-            meta.name = name or meta.name
+        # create pd.Series from index and meta if provided
+        _meta = meta if index is None else pd.Series(data=meta, index=index)
```

Comment:
I don't think I see any need here to use `_meta`, perhaps just use `meta`?

Reply:
this is just for `_add_meta_column` right? I think that can be ameliorated with the suggested change to that function below.